### PR TITLE
frontend: Add VRR option within Gamescope

### PR DESCRIPTION
**Pull request by TheEvilSkeleton**
# Description
This adds a VRR option within Gamescope for users who use a desktop that supports VRR.

If the session uses X11 or a Wayland compositor that supports VRR, then this feature will be available to the user and will be applied when the user runs Gamescope.

Closes https://github.com/bottlesdevs/Bottles/issues/2178.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce.
- [x] Locally

### CODE REVIEW

**Comment by TheEvilSkeleton** [comment]
This breaks since I get an error:

```
Traceback (most recent call last):
  File "/app/share/bottles/bottles/frontend/views/bottle_preferences.py", line 506, in __show_gamescope_settings
    new_window = GamescopeDialog(
  File "/app/share/bottles/bottles/frontend/windows/gamescope.py", line 63, in __init__
    self.__update(config)
  File "/app/share/bottles/bottles/frontend/windows/gamescope.py", line 94, in __update
    self.switch_vrr.set_active(parameters["gamescope_vrr"])
KeyError: 'gamescope_vrr'
```

I'll open a new MR that migrates to `.get()`.

**Comment by TheEvilSkeleton** [comment]
This is how it looks like (ignore underline bug):

image

File: bottles/frontend/windows/gamescope.py [hunk 1]
```diff
@@ -54,6 +58,17 @@ def __init__(self, window, config, **kwargs):
...
         self.__update(config)
 
+        # Set VRR row to visible if the compositor supports VRR
+        if vrr_available:
+            self.row_vrr.set_visible(True)
+
+            if DisplayUtils.display_server_type() == "wayland":
+                warning = _("This feature depends on the version of the compositor.")
+                learn_more = _("Learn more about variable refresh rate")
+                link = f"<a href=\"LINK\">{learn_more}</a>"
```
Comment (TheEvilSkeleton) [inline bottles/frontend/windows/gamescope.py:68]: This needs to be updated. @mirkobrombin can we create a page for VRR?

Comment (TheEvilSkeleton) [reply bottles/frontend/windows/gamescope.py:68]: I guess this works. https://wiki.archlinux.org/title/Variable_refresh_rate

Comment (mirkobrombin) [reply bottles/frontend/windows/gamescope.py]: I think ArchWiki cover it better.

**Comment by TheEvilSkeleton** [comment]
Blocking this MR, as we agreed on solely fixing bugs.

File: bottles/backend/globals.py [hunk 2]
```diff
@@ -52,6 +53,19 @@ def is_vkbasalt_available():
...
+                            "sway",
+                            "plasma",
+                          ]
+
+    desktop = environ.get("DESKTOP_SESSION")
+    check_wayland_session = True if DisplayUtils.display_server_type() == "wayland" else False
+    check_x11_session = True if DisplayUtils.display_server_type() == "x11" else False
+
+    return True if (desktop in vrr_wayland_support and check_wayland_session) or check_x11_session else False
+
```
Comment (orowith2os) [inline bottles/backend/globals.py:68]: Let's cut this out, it's going to be unmanageable if this is how it's done in the future.

File: bottles/frontend/windows/gamescope.py [hunk 3]
```diff
@@ -54,6 +58,17 @@ def __init__(self, window, config, **kwargs):
...
+        if vrr_available:
+            self.row_vrr.set_visible(True)
+
+            if DisplayUtils.display_server_type() == "wayland":
+                warning = _("This feature depends on the version of the compositor.")
+                learn_more = _("Learn more about variable refresh rate")
+                link = f"<a href=\"https://wiki.archlinux.org/title/Variable_refresh_rate#Wayland_configuration\">{learn_more}</a>"
+                message = f"{self.row_vrr.get_subtitle()} {warning} {link}"
+                self.row_vrr.set_subtitle(message)
+
```
Comment (orowith2os) [inline bottles/frontend/windows/gamescope.py:71]: Let's just cut this out, and say "Enable VRR if available." Far cleaner, doesn't limit us to checks for specific compositors, and doesn't make unnecessary checks.

**Review by orowith2os — CHANGES_REQUESTED** [review]

**Comment by TheEvilSkeleton** [comment]
We'll revisit it when Bottles is stable enough https://github.com/bottlesdevs/Bottles/pull/2373#issuecomment-1368988399

**Comment by TheEvilSkeleton** [comment]
Closing this as I won't be working on it anymore

**Pull request closed without merging**